### PR TITLE
feat: add ruby-foreman-rescue 2.0.1 plugin

### PR DESCRIPTION
**Pull request by marcdeop**
This PR adds ruby-foreman-rescue to the deb building process

**NOTE**: I created this PR as #3946 wouldn't work due to CI issues

**NOTE**: this work is thanks to [Erasys GmbH](https://www.erasys.de)

### CODE REVIEW

**Comment by dgoetz** [comment]
@marcdeop Why version 1.0.0 and not 2.0.0?
@timogoebel Would it make sense to release a 2.0.1 with the seeding error fixed before adding packages to Debian?

**Comment by marcdeop** [comment]
> @marcdeop Why version 1.0.0 and not 2.0.0?

To be honest, I forgot about the 2.0.0 because our Foreman instance is still running version 1.19.X and I was going to install the 1.0.0 version of the plugin.

AFAIK, version 1.0.0 would work with newer versions of Foreman as well. Regardless, if you see the need for it I will upgrade the version ;-)

**Comment by tbrisker** [comment]
[test packaging]

File: plugins/ruby-foreman-rescue/debian/install [hunk 1]
```diff
@@ -0,0 +1,2 @@
+foreman_rescue.rb              usr/share/foreman/bundler.d
+cache                          usr/share/foreman/vendor
```
Comment (tbrisker) [inline plugins/ruby-foreman-rescue/debian/install]: Looks like this is failing the packaging: 
```
17:38:13 cp cache/* /usr/share/foreman/vendor/cache/
17:38:13 cp: cannot create regular file '/usr/share/foreman/vendor/cache/': No such file or directory
```

Comment (marcdeop) [reply plugins/ruby-foreman-rescue/debian/install]: I obviously did something wrong. I will get back to you ASAP

Comment (tbrisker) [reply plugins/ruby-foreman-rescue/debian/install]: no rush :)

**Comment by marcdeop** [comment]
[test packaging]

**Comment by tbrisker** [comment]
ok to test

**Comment by tbrisker** [comment]
[test packaging]

**Comment by marcdeop** [comment]
[test packaging]

File: plugins/ruby-foreman-rescue/debian/rules [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+# dh-make output file, you may use that output file without restriction.
+# This special exception was added by Craig Small in version 0.37 of dh-make.
+
+# Uncomment this to turn on verbose mode.
+#export DH_VERBOSE=1
+
+PLUGIN = foreman_rescue
+
+build:
+	cp cache/* /usr/share/foreman/vendor/cache/
```
Comment (tbrisker) [inline plugins/ruby-foreman-rescue/debian/rules]: Looks like this step is failing

File: plugins/ruby-foreman-rescue/debian/rules [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+		/usr/bin/foreman-ruby /usr/bin/bundle install --local && \
+		/usr/bin/foreman-ruby /usr/bin/bundle exec rake plugin:assets:precompile[$(PLUGIN)] RAILS_ENV=production && \
+		/usr/bin/foreman-ruby /usr/bin/bundle exec rake db:migrate RAILS_ENV=development && \
+		/usr/bin/foreman-ruby /usr/bin/bundle exec rake plugin:apipie:cache[$(PLUGIN)] cache_part=resources \
+			OUT=/var/lib/foreman/public/apipie-cache/plugin/$(PLUGIN) RAILS_ENV=development \
+	)
+	GEM_PATH=$$(cd /usr/share/foreman && /usr/bin/foreman-ruby /usr/bin/bundle show $(PLUGIN)) && \
+		cp -rp $${GEM_PATH}/public/assets/$(PLUGIN) ./
+	[ -e apipie-cache ] || mkdir apipie-cache/
+	cp -rp /var/lib/foreman/public/apipie-cache/plugin/$(PLUGIN) ./apipie-cache/
```
Comment (tbrisker) [inline plugins/ruby-foreman-rescue/debian/rules]: There is no api in the plugin so i think we don't need any of the apipie steps

File: plugins/ruby-foreman-rescue/debian/copyright [hunk 4]
```diff
@@ -0,0 +1,32 @@
+Format: http://www.debian.org/doc/packaging-manuals/copyright-format/1.0/
+Upstream-Name: ruby-foreman-rescue
+Source: https://github.com/dm-drogeriemarkt/foreman_rescue
+
+
+Files: *
+Copyright: 2019 Marc Deop <marc.deop@erasys.de>
```
Comment (tbrisker) [inline plugins/ruby-foreman-rescue/debian/copyright]: I think this should be the copyright owner of the plugin itself, but not sure

File: plugins/ruby-foreman-rescue/debian/rules (hunk 2 above)
Comment (marcdeop) [reply plugins/ruby-foreman-rescue/debian/rules]: Fixed this adding proper dependencies.

File: plugins/ruby-foreman-rescue/debian/rules (hunk 3 above)
Comment (marcdeop) [reply plugins/ruby-foreman-rescue/debian/rules]: Cleaned this up

**Comment by marcdeop** [comment]
@dgoetz I upgraded the plugin to version 2.0.0

**Comment by mmoll** [comment]
let's wait for a 2.0.1 with the seed fix

**Comment by timogoebel** [comment]
@marcdeop: I've just released foreman_rescue 2.0.1 that includes the seed fix.

**Comment by marcdeop** [comment]
Is there anything else that needs to be done to push this forward?

Is the `.deb` file anywhere so I could try it?

**Comment by mmoll** [comment]
@marcdeop yes in https://stagingdeb.theforeman.org/pool/plugins/marcdeop/r/ruby-foreman-rescue/
I'd like to wait for 1.23-RC1 to arrive (should be really soon), so I don't have to rebuild the plugin for 1.23 later.

**Comment by marcdeop** [comment]
> @marcdeop yes in https://stagingdeb.theforeman.org/pool/plugins/marcdeop/r/ruby-foreman-rescue/
> I'd like to wait for 1.23-RC1 to arrive (should be really soon), so I don't have to rebuild the plugin for 1.23 later.

Sure, no problem, I was making sure that everything is ready.

If you want I can rebase the branch to keep up with the changes.

**Comment by mmoll** [comment]
It builds, so let's merge this and wait for user reports. :tada:

**Comment by mmoll** [comment]
Cherry-picked all the way down to 1.20, danke @marcdeop!

**Pull request merged**